tests/bluetooth/bsim_bt/bsim_test_notify/src/gatt_client_test.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+
+#include "common.h"
+
+CREATE_FLAG(flag_is_connected);
+CREATE_FLAG(flag_is_encrypted);
+CREATE_FLAG(flag_discover_complete);
+CREATE_FLAG(flag_subscribed);
+
+static struct bt_conn *g_conn;
+static uint16_t chrc_handle;
+static uint16_t long_chrc_handle;
+static struct bt_uuid *test_svc_uuid = TEST_SERVICE_UUID;
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		FAIL("Failed to connect to %s (%u)\n", addr, err);
+		return;
+	}
+
+	printk("Connected to %s\n", addr);
+
+	g_conn = conn;
```

Comment:
`g_conn` is supplied to `bt_conn_le_create` and should already be a valid pointer. 

```suggestion
```